HoneyDo.Web/ClientApp/src/pages/Tasks.tsx

```diff
@@ -85,6 +85,21 @@ class Tasks extends Component<TasksProps, TasksState> {
     this.updateExistingTask(updatedTask);
   };
 
+  handleDelete = async (task: Task) => {
+    const { deleteTask }: TaskContextData = this.context;
+    const isDeleted = await deleteTask(task);
+    if (!isDeleted) {
+      return;
```

Comment:
Shouldn't this reject it rather than return?

Reply:
I do like the defensive programming though!!

Reply:
That's essentially what this is doing.  It's returning before any additional logic can be performed.  An error message should show at this point, but we don't have that yet.

Reply:
I thought this was a promise and by returning, you are resolving it? I thought I saw this as a type of `Promise <Boolean>`. Could be lying though.

Reply:
This doesn’t return anything.

Reply:
If it's not supposed to, that's fine. Here's where I thought it should reject it:
in `TaskContext.tsx`:
```js
export interface IDeleteTask {
  (task: Task): Promise<boolean>;
}
```
This is set to return a Promise with a boolean value?
Then I saw this:
```js
export const TaskContext = React.createContext<TaskContextData>({
  // ....
  deleteTask: async () => false
  });
});
```
deleteTask is a async function that returns false. If just returning is the same thing, I can resolve this and I'm good. Otherwise I may not be following correctly?

Reply:
You're talking about 2 different files and 2 different functions.  The function this comment is on calls the `deleteTask` function in the `TaskContext.tsx`, the function this comment is on **does not** need to return anything.  The `deleteTask` in `TaskContext.tsx` correctly implements a return of a `boolean`.